simulator.py

```diff
@@ -267,7 +267,7 @@ def state(self):
         return [int(self.is_susceptible), int(self.is_exposed), int(self.is_infectious), int(self.is_removed)]
 
     def assert_state_changes(self):
-        next_state = {0:1, 1:2, 2:0}
+        next_state = {0:1, 1:2, 2:3}
```

Comment:
I was sloppy when I coded this. basically, an infected person can either be dead or be back to susceptible. so the state transition can either be `2 --> 3` or `2 --> 0`. I think we can use a list and assert the membership based on `self.recovered_timestamp`.